### PR TITLE
[Merged by Bors] - chore(PartENat): golf and improve `ofNat` support

### PR DESCRIPTION
**Pull request by timotree3**
This PR adds simp lemmas for `OfNat.ofNat n : PartENat`, `0 : PartENat`, and `1 : PartENat` in every place where there was a simp lemma for `((n : ℕ) : PartENat)`. This is necessary for simp confluence in the presence of lemmas such as `Nat.cast_ofNat`. In addition, instances for `CharZero` and `ZeroLEOneClass` are provided so that the lemmas from `Data/Nat/Cast/Order.lean` will apply, golfing some proofs.

---

- [ ] depends on: #8129 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by eric-wieser** [comment]
!bench

File: Mathlib/Data/Nat/PartENat.lean [hunk 1]
```diff
@@ -108,16 +108,32 @@ theorem some_eq_natCast (n : ℕ) : some n = n :=
...
+theorem dom_ofNat (x : ℕ) [x.AtLeastTwo] : (no_index (OfNat.ofNat x : PartENat)).Dom :=
+  trivial
+
+@[simp]
+theorem dom_zero : ((0 : PartENat)).Dom :=
+  trivial
+
+@[simp]
+theorem dom_one : ((1 : PartENat)).Dom :=
+  trivial
```
Comment (eric-wieser) [inline Mathlib/Data/Nat/PartENat.lean:135]:
```suggestion
@[simp]
theorem dom_zero : (0 : PartENat).Dom :=
  trivial

@[simp]
theorem dom_one : (1 : PartENat).Dom :=
  trivial
```

File: Mathlib/LinearAlgebra/Orientation.lean [hunk 2]
```diff
@@ -115,14 +115,14 @@ theorem Orientation.reindex_symm (e : ι ≃ ι') :
 end Reindex
 
 /-- A module is canonically oriented with respect to an empty index type. -/
-instance (priority := 100) IsEmpty.oriented [Nontrivial R] [IsEmpty ι] : Module.Oriented R M ι where
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/Orientation.lean:118]: Can you explain in the PR description how you managed to eliminate these?

Comment (timotree3) [reply Mathlib/LinearAlgebra/Orientation.lean:118]: I got an unused variable warning on these so removed them. Turns out they're redundant on master too. Added #8129 as a dependency PR.

File: Mathlib/Data/Nat/PartENat.lean [hunk 3]
```diff
@@ -568,6 +605,15 @@ theorem toWithTop_zero' {h : Decidable (0 : PartENat).Dom} : toWithTop 0 = 0 :=
   convert toWithTop_zero
 #align part_enat.to_with_top_zero' PartENat.toWithTop_zero'
 
+theorem toWithTop_one :
+    have : Decidable (1 : PartENat).Dom := someDecidable 1
```
Comment (eric-wieser) [inline Mathlib/Data/Nat/PartENat.lean:609]:
```suggestion
    haveI : Decidable (1 : PartENat).Dom := someDecidable 1
```
to inline it

File: Mathlib/Data/Nat/PartENat.lean [hunk 4]
```diff
@@ -729,6 +788,15 @@ theorem withTopEquiv_symm_zero : withTopEquiv.symm 0 = 0 :=
   rfl
 #align part_enat.with_top_equiv_symm_zero PartENat.withTopEquiv_symm_zero
 
+@[simp]
+theorem withTopEquiv_symm_one : withTopEquiv.symm 1 = 1 :=
+  rfl
+
+@[simp]
+theorem withTopEquiv_symm_ofNat (n : Nat) [n.AtLeastTwo] :
```
Comment (eric-wieser) [inline Mathlib/Data/Nat/PartENat.lean:796]: I don't see `withTopEquiv_symm_natCast` above; is it missing?

Comment (eric-wieser) [reply Mathlib/Data/Nat/PartENat.lean:796]: Ah, it's called `coe`

File: Mathlib/Data/Nat/PartENat.lean (hunk 3 above)
Comment (eric-wieser) [reply Mathlib/Data/Nat/PartENat.lean:609]: Nevermind, I see you're just matching the existing pattern

**Comment by eric-wieser** [comment]
bors d+

Pleaes merge once CI passes.

Thanks, and sorry for letting this sit for a little.

**Comment by eric-wieser** [comment]
bors merge

**Comment by eric-wieser** [comment]
!bench

**Pull request closed without merging**